### PR TITLE
Implemented the sign up and login functionality for the backend 

### PR DESCRIPTION
**Pull request by suresh-02**

### CODE REVIEW

File: backend/handlers/authHandler.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+        "sub": user.ID,
+        "exp": time.Now().Add(time.Hour).Unix(),
+    })
+
+    tokenString, err := token.SignedString([]byte(os.Getenv("SECRET")))
+    if err != nil {
+        http.Error(w, "Error generating JWT token", http.StatusInternalServerError)
+        return
+    }
```
[the remark this answers is not in the document]
Comment (suresh-02) [reply backend/handlers/authHandler.go]: Resolved in [e068f8f](https://github.com/suresh-02/Iragu-booking/pull/2/commits/e068f8f83bec8c1c1c9060747f2bb66a062c7649)

File: backend/handlers/authHandler.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		return
+	}
+
+	fmt.Println(user.Password)
+	// http.Error(w, user.Password, http.StatusAccepted)
+	err := bcrypt.CompareHashAndPassword([]byte(user.Password), []byte(loginBody.Password))
+	if err != nil {
+		http.Error(w, "Invalid password", http.StatusForbidden)
+		return
+	}
```
[the remark this answers is not in the document]
Comment (suresh-02) [reply backend/handlers/authHandler.go]: Resolved in [e068f8f](https://github.com/suresh-02/Iragu-booking/pull/2/commits/e068f8f83bec8c1c1c9060747f2bb66a062c7649)

File: backend/handlers/authHandler.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	json.NewDecoder(r.Body).Decode(&loginBody)
+
+	var user models.UserCreds
+
+	database.DB.First(&user, "email=?", loginBody.Email)
+
+	if user.ID == 0 {
+		http.Error(w, "No user found", http.StatusBadRequest)
+		return
+	}
```
[the remark this answers is not in the document]
Comment (suresh-02) [reply backend/handlers/authHandler.go]: Resolved in [e068f8f](https://github.com/suresh-02/Iragu-booking/pull/2/commits/e068f8f83bec8c1c1c9060747f2bb66a062c7649)

File: backend/handlers/authHandler.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		Password: string(hashedPassword),
+		Email:    signupBody.Email,
+	}
+
+	// Save the user to the database
+	result := database.DB.Create(&signupUser)
+	if result.Error != nil {
+		http.Error(w, "Error creating user", http.StatusBadRequest)
+		return
+	}
```
[the remark this answers is not in the document]
Comment (suresh-02) [reply backend/handlers/authHandler.go]: Resolved in [e068f8f](https://github.com/suresh-02/Iragu-booking/pull/2/commits/e068f8f83bec8c1c1c9060747f2bb66a062c7649)

File: backend/handlers/authHandler.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	// Hash the password
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(signupBody.Password), 10)
+	if err != nil {
+		http.Error(w, "Failed to create hashed password", http.StatusInternalServerError)
+		return
+	}
```
[the remark this answers is not in the document]
Comment (suresh-02) [reply backend/handlers/authHandler.go]: Resolved in [e068f8f](https://github.com/suresh-02/Iragu-booking/pull/2/commits/e068f8f83bec8c1c1c9060747f2bb66a062c7649)

File: backend/handlers/authHandler.go [hunk 6]
```diff
@@ -45,14 +45,14 @@ func SignUp(w http.ResponseWriter, r *http.Request) {
...
-		http.Error(w, "Error creating user", http.StatusBadRequest)
+		http.Error(w, "Invalid Email or Password", http.StatusInternalServerError)
 		return
 	}
 
 	// Send a success response
 	w.WriteHeader(http.StatusCreated)
 	json.NewEncoder(w).Encode(map[string]string{
-		"message": "User created successfully",
+		"message": "User created susfullycces",
```
Comment (Ramanan-Bala) [inline backend/handlers/authHandler.go:55]: enna bro spelling

File: backend/handlers/authHandler.go [hunk 7]
```diff
@@ -70,34 +70,38 @@ func Login(w http.ResponseWriter, r *http.Request) {
 	database.DB.First(&user, "email=?", loginBody.Email)
 
 	if user.ID == 0 {
-		http.Error(w, "No user found", http.StatusBadRequest)
+		http.Error(w, "Invalid Email or Password", http.StatusInternalServerError)
```
Comment (Ramanan-Bala) [inline backend/handlers/authHandler.go:73]: 404 ehh correct tha

File: backend/handlers/authHandler.go [hunk 8]
```diff
@@ -70,34 +70,38 @@ func Login(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
 	fmt.Println(user.Password)
 	// http.Error(w, user.Password, http.StatusAccepted)
 	err := bcrypt.CompareHashAndPassword([]byte(user.Password), []byte(loginBody.Password))
 	if err != nil {
-		http.Error(w, "Invalid password", http.StatusForbidden)
+		http.Error(w, "Invalid Email or Password", http.StatusForbidden)
```
Comment (Ramanan-Bala) [inline backend/handlers/authHandler.go:81]: 404

File: backend/handlers/authHandler.go [hunk 9]
```diff
@@ -70,34 +70,38 @@ func Login(w http.ResponseWriter, r *http.Request) {
...
-    json.NewEncoder(w).Encode(map[string]string{
-        "token": tokenString,
-    })
+		"sub": user.ID,
+		"exp": time.Now().Add(time.Hour).Unix(),
+	})
+
+	tokenString, err := token.SignedString([]byte(os.Getenv("SECRET")))
+	if err != nil {
+		http.Error(w, "Invalid Email or Password", http.StatusInternalServerError)
```
Comment (Ramanan-Bala) [inline backend/handlers/authHandler.go:94]: 404

File: backend/handlers/authHandler.go [hunk 10]
```diff
@@ -31,7 +31,7 @@ func SignUp(w http.ResponseWriter, r *http.Request) {
 	// Hash the password
 	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(signupBody.Password), 10)
 	if err != nil {
-		http.Error(w, "Failed to create hashed password", http.StatusInternalServerError)
+		http.Error(w, "Invalid Email or Password", http.StatusInternalServerError)
```
Comment (Ramanan-Bala) [inline backend/handlers/authHandler.go:34]: 404

File: backend/handlers/authHandler.go [hunk 11]
```diff
@@ -45,14 +45,14 @@ func SignUp(w http.ResponseWriter, r *http.Request) {
 	// Save the user to the database
 	result := database.DB.Create(&signupUser)
 	if result.Error != nil {
-		http.Error(w, "Error creating user", http.StatusBadRequest)
+		http.Error(w, "Invalid Email or Password", http.StatusInternalServerError)
```
Comment (Ramanan-Bala) [inline backend/handlers/authHandler.go:48]: 404

File: backend/main.go [hunk 12]
```diff
@@ -1,8 +1,34 @@
...
+	// function to connect to the database
+	database.Connect()
+	// function to auto migrate and sync schema changes with the database
+	database.SyncDatabase()
+
+	// routes
+	// Initialize the router
+
+	// Define routes
+	router.HandleFunc("/newUser", handlers.SignUp).Methods(http.MethodPost)
```
Comment (Ramanan-Bala) [inline backend/main.go:26]: register

**Pull request merged**